src/scala/com/github/johnynek/bazel_deps/DepsModel.scala

```diff
@@ -159,12 +172,20 @@ case class ArtifactOrProject(asString: String) {
     val str = sp.asString
     str match {
       case "" => this
-      case _ => ArtifactOrProject(s"$asString-$str")
+      case _ => ArtifactOrProject(s"$artifact-$str", packaging, classifier)
     }
   }
 }
 object ArtifactOrProject {
   implicit val ordering: Ordering[ArtifactOrProject] = Ordering.by(_.asString)
+
+  def apply(str: String): ArtifactOrProject = {
+    str.split(":") match {
+      case Array(a, p, c) => ArtifactOrProject(a, p, Some(c))
+      case Array(a, p) => ArtifactOrProject(a, p, None)
+      case Array(a) => ArtifactOrProject(a, "jar", None)
```

Comment:
I see `"jar"` repeated many times as the default packaging.  Replace them with a `val defaultPackaging = "jar"` for readability?